ConfigurationSystem/scripts/dirac-admin-voms-sync.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+########################################################################
+# File :   dirac-admin-voms-sync
+# Author : Andrei Tsaregorodtsev
+########################################################################
+"""
+  Synchronize VOMS user data with the DIRAC Registry
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC import gLogger, exit as DIRACExit, S_OK
+from DIRAC.Core.Base import Script
+from DIRAC.ConfigurationSystem.Client.VOMS2CSSyncronizer import VOMS2CSSynchronizer
+
+dryRun = False
+
+
+def setDryRun(value):
+  global dryRun
+  dryRun = True
+  return S_OK()
+
+
+voName = None
+
+
+def setVO(value):
+  global voName
+  voName = value
+  return S_OK()
+
+
+Script.registerSwitch("V:", "vo=", "VO name", setVO)
+Script.registerSwitch("D", "dryRun", "Dry run", setDryRun)
+Script.setUsageMessage('\n'.join([__doc__.split('\n')[1],
+                                  'Usage:',
+                                  '  %s [option|cfgfile] ...' % Script.scriptName]
+                                 ))
+
+Script.parseCommandLine(ignoreErrors=True)
+
+vomsSync = VOMS2CSSynchronizer(voName)
+result = vomsSync.syncCSWithVOMS()
+if not result['OK']:
+  gLogger.error("Failed to synchronize user data")
+  DIRACExit(-1)
+
+resultDict = result['Value']
+newUsers = resultDict.get("NewUsers", [])
+modUsers = resultDict.get("ModifiedUsers", [])
+delUsers = resultDict.get("DeletedUsers", [])
+susUsers = resultDict.get("SuspendedUsers", [])
+gLogger.notice("\nUser results: new %d, modified %d, deleted %d, new/suspended %d" %
+               (len(newUsers), len(modUsers), len(delUsers), len(susUsers)))
+
+for msg in resultDict["AdminMessages"]["Info"]:
+  gLogger.notice(msg)
+
+csapi = resultDict.get("CSAPI")
+if csapi and csapi.csModified:
+  if dryRun:
+    gLogger.notice("There are changes to Registry ready to commit, skipped because of dry run")
+  else:
+    yn = raw_input("There are changes to Registry ready to commit, do you want to proceed [default yes] [yes|no]:")
```

Comment:
should be `[Y|n]` given the check in the next line (upper case marking the default choice). Though shouldn't the default be "no" ?

Reply:
The check allows both Y and y and I will make it allowing also Yes and yes

Reply:
I knew that both Y and y were working, but some command line tools use capitalisation to indicate the default value, thus `[Y|n]`

Still my question is if the default here shouldn't be NO do no commit... so  `[y|N]`

Reply:
This is the matter of taste, I think. If you do not make a dry run, then you most probably intend to commit the changes. So, I would prefer the default to be Yes.